src/main/scala/TraceIO.scala

```diff
@@ -4,51 +4,105 @@ import chisel3._
 import chisel3.util._
 
 import freechips.rocketchip.subsystem.{BaseSubsystem, HasTiles}
-import freechips.rocketchip.config.{Field, Config}
+import freechips.rocketchip.config.{Field, Config, Parameters}
 import freechips.rocketchip.diplomacy.{LazyModule, AddressSet, LazyModuleImp, BundleBridgeNexus}
 import freechips.rocketchip.tilelink.{TLRAM}
-import freechips.rocketchip.rocket.TracedInstruction
+import freechips.rocketchip.rocket.{TracedInstruction}
 import freechips.rocketchip.util._
+import freechips.rocketchip.tile.{BaseTile}
+import freechips.rocketchip.diplomacy.{BundleBridgeSource, BundleBroadcast, BundleBridgeNexus}
 
-case class TracePortParams(
-  print: Boolean = false
-)
-
-object TracePortKey extends Field[Option[TracePortParams]](None)
+//***************************************************************************
+// Extended Trace Instruction Utilities:
+// Used to map TracedInstructions to their Extended version and
+// used to connect the TracerV or Dromajo bridges (in FireSim and normal sim)
+//***************************************************************************
 
-
-case class TracedInstructionWidths(iaddr: Int, insn: Int, cause: Int, tval: Int)
+case class TracedInstructionWidths(iaddr: Int, insn: Int, wdata: Int, cause: Int, tval: Int)
 
 object TracedInstructionWidths {
+  def apply(tI: ExtendedTracedInstruction): TracedInstructionWidths =
+    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, tI.wdata.getWidth, tI.cause.getWidth, tI.tval.getWidth)
+
+  // note: the wdata is not 0 here since we can't deal with 0 width wires
   def apply(tI: TracedInstruction): TracedInstructionWidths =
-    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, tI.cause.getWidth, tI.tval.getWidth)
+    TracedInstructionWidths(tI.iaddr.getWidth, tI.insn.getWidth, 1, tI.cause.getWidth, tI.tval.getWidth)
+}
+
+class ExtendedTracedInstruction(implicit p: Parameters) extends TracedInstruction {
+  val wdata = UInt(xLen.W)
+}
+
+object ExtendedTracedInstruction {
+  def apply(tI: TracedInstruction): ExtendedTracedInstruction = {
+    val temp = Wire(new ExtendedTracedInstruction()(tI.p))
+    temp.valid := tI.valid
+    temp.iaddr := tI.iaddr
+    temp.insn := tI.insn
+    temp.wdata := 0.U
+    temp.priv := tI.priv
+    temp.exception := tI.exception
+    temp.interrupt := tI.interrupt
+    temp.cause := tI.cause
+    temp.tval := tI.tval
+
+    temp
+  }
+
+  def fromVec(tis: Vec[TracedInstruction]): Vec[ExtendedTracedInstruction] = {
+    val tempVec = tis.map(insn => Wire(new ExtendedTracedInstruction()(insn.p)))
+
+    tempVec.zip(tis).foreach({ case (ext, non_ext) =>
```

Comment:
Why can't this apply the previous function?

Reply:
I think it can. Ill go ahead and update.